### PR TITLE
[feat](Nereids): add struct info map in group

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

add struct info map in group for extracting struct info according mv

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/test/java/org/apache/doris/nereids/util/PlanChecker.java [hunk 1]
```diff
@@ -121,6 +122,7 @@ public PlanChecker parse(String sql) {
     public PlanChecker analyze() {
         this.cascadesContext.newAnalyzer().analyze();
         this.cascadesContext.toMemo();
+        InitMaterializationContextHook.INSTANCE.initMaterializationContext(this.cascadesContext);
```
Comment (seawinde) [inline fe/fe-core/src/test/java/org/apache/doris/nereids/util/PlanChecker.java:125]: maybe this should judge if the `enable_materialized_view_rewrite` is true or not, and then set the  `enable_materialized_view_rewrite` in the test respectively

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+/**
+ * Representation for group in cascades optimizer.
+ */
+public class StructInfoMap {
+    private final Map<BitSet, GroupExpression> groupExpressionMap = new HashMap<>();
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java:37]: If self join, how we can represent it

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    }
+
+    private Set<BitSet> cartesianProduct(List<Set<BitSet>> childrenTableMap) {
+        Set<BitSet> result = new HashSet<>();
+        cartesianProductHelper(childrenTableMap, 0, new BitSet(), result);
+        return result;
+    }
+
+    private void cartesianProductHelper(List<Set<BitSet>> childrenTableMap, int currentIndex,
+            BitSet currentTableMap, Set<BitSet> res) {
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java:115]: guava has `com.google.common.collect.Sets#cartesianProduct(java.util.Set<? extends B>...)` maybe helpful

File: fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java (hunk 2 above)
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/memo/StructInfoMap.java:37]: don't represent it. Right now, we just use bitmap to support the basic functions

**Comment by XieJiann** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Review by jackwener — APPROVED** [review]

**Pull request merged**